runtime/src/main/java/org/corfudb/runtime/object/transactions/AbstractTransactionalContext.java

```diff
@@ -248,6 +251,17 @@ public void syncWithRetryUnsafe(VersionLockedObject vlo,
      */
     public abstract void addTransaction(AbstractTransactionalContext tc);
 
+    /**
+     * Add an object that needs extra processing right before commit happens
+     *
+     * @param preCommitListener The context of the object that needs extra processing
+     *                         along with its lambda.
+     */
+    public abstract void addPreCommitListener(PreCommitListener preCommitListener);
```

Comment:
I'm wondering if this is correct in case of nested transactions. In that case we would merge the read/write set in the child transaction's context into parent's context, however the preCommitListeners seems not doing this.

Reply:
I thought about this - that is the reason I always pick the RootContext to store these preCommitListeners. As you said nesting transactions will keep merging their writeSets() anyway which is is unnecessary because only the address of the parent transaction matters anyway.